base/bucket_gocb.go

```diff
@@ -876,8 +884,32 @@ func (bucket CouchbaseBucketGoCB) GetDDoc(docname string, into interface{}) erro
 }
 
 func (bucket CouchbaseBucketGoCB) PutDDoc(docname string, value interface{}) error {
-	LogPanic("Unimplemented method: PutDDoc()")
-	return nil
+
+	// Get bucket manager.  Relies on existing auth settings for bucket.
+	username, password := bucket.GetBucketCredentials()
+	manager := bucket.Bucket.Manager(username, password)
+	if manager == nil {
+		return fmt.Errorf("Unable to obtain manager for bucket %s - cannot initialize view", bucket.GetName())
+	}
```

Comment:
What about instead of "cannot initialize view" rewording this to "Cannot PUT design doc"?